fea-rs/src/compile/compile_ctx.rs

```diff
@@ -1362,71 +1362,85 @@ impl<'a, F: FeatureProvider, V: VariationInfo> CompilationCtx<'a, F, V> {
+                for name_spec in feature_name.statements() {
+                    let resolved = self.resolve_name_spec(&name_spec);
+                    // in the case of duplicate names, last writer wins, see
+                    // https://github.com/googlefonts/fontc/issues/1019
+                    let key = resolved.key();
+                    names.insert(key, resolved);
+                }
+                continue;
             }
+            self.resolve_statement(item);
         }
         if !names.is_empty() {
-            self.features.stylistic_sets.insert(tag, names);
-        }
-        for item in feature
-            .statements()
-            .filter(|node| node.kind() != Kind::FeatureNamesKw)
-        {
-            self.resolve_statement(item);
+            self.features
+                .stylistic_sets
+                .insert(tag, names.into_values().collect());
         }
     }
 
     fn resolve_character_variant_feature(&mut self, tag: Tag, feature: &typed::Feature) {
-        if let Some(cv_params) = feature.character_variant_params() {
-            let mut params = CvParams::default();
-            if let Some(node) = cv_params.feat_ui_label_name() {
-                params.feat_ui_label_name = node
-                    .statements()
-                    .map(|x| self.resolve_name_spec(&x))
-                    .collect();
-            }
-            if let Some(node) = cv_params.feat_tooltip_text_name() {
-                params.feat_ui_tooltip_text_name = node
-                    .statements()
-                    .map(|x| self.resolve_name_spec(&x))
-                    .collect();
-            }
-            if let Some(node) = cv_params.sample_text_name() {
-                params.sample_text_name = node
-                    .statements()
-                    .map(|x| self.resolve_name_spec(&x))
-                    .collect();
-            }
-            if let Some(node) = cv_params.sample_text_name() {
-                params.sample_text_name = node
-                    .statements()
-                    .map(|x| self.resolve_name_spec(&x))
-                    .collect();
-            }
-            for node in cv_params.param_ui_label_name() {
-                params.param_ui_label_names.push(
-                    node.statements()
-                        .map(|x| self.resolve_name_spec(&x))
-                        .collect(),
-                );
-            }
-            for c in cv_params.characters() {
-                params.characters.push(c.value().parse_char().unwrap());
+        let mut seen_params = false;
+        for item in feature.statements() {
+            if let Some(cv_params) = typed::CvParameters::cast(item) {
+                if seen_params {
+                    self.warning(
+                        cv_params.range(),
+                        "Duplicate cvParameters block will be ignored. \
+                        This may technically be okay but is currently unhandled.",
```

Comment:
Couple concerns with this message:

1. We don't appear to check for duplicate, merely that we've seen any other cvParameters block
1. We should clarify our handling, e.g. note that only the first one is used
1. "This may technically be okay but is currently unhandled." suggests we don't know if its ok or not. Could we check other implementations and make a stronger claim? 

For example, "Ignoring cvParameters, we already saw one. Multiple cvParameters are not allowed." or "Ignoring cvParameters, we already saw one. Multiple cvParameters not yet supported." would be more useful to me as user.